### ISSUE TITLE
Add greenness workflow to SciF solution

### ISSUE THREAD

**Issue by Chris-Schnaufer** [AgPipeline/computing-pipeline#227]
**Task to do**
Add greenness index workflow to existing SciF solution. Investigate if it's possible to run one workflow and use that output as input to the other (canopy cover vs. greenness index for example) if the workflows are cloned vs. reconstructed.

If reconstructing the workflows are needed due to inability to leverage previous runs on another algorithm (such as re-using already clipped images), break the workflow apart into: **preceding steps** -> plot clip app -> canopy cover app _or_ **preceding steps** -> plot clip app -> greenness app

**Reason**
Want to have more than one algorithm available

**Result**
Able to easily generate canopy cover and greenness values

**Notes**
The task is to investigate the best approach; choosing between:
* cloning and modifying canopy cover workflow for greenness index
* break up workflow into plot clip app, canopy cover app, greenness app

The user would then need to run each of these apps sequentially

**Comment by Chris-Schnaufer** [comment]
There are several ways of addressing this. A preliminary approach that hopefully incorporates previous discussions is:
1. Minimize the Makeflow workflows built into the SciF container (for example, keep Orthomosaic to plotclip as one step, but no other Makeflow workflows)
2. Provide an app for running a non-SciF Makeflow files (a generic Makeflow entrypoint) running user workflows (security issues can arise from this since we have no control over what's provided)
3. Provide Makeflow files that can be used outside of the Scif image, and others that can be used for the generic app, for supported workflows & document how they can be extended
4. Add new apps as needed (generating a list of plot level files clipped from an Orthomosaic)

**Comment by Chris-Schnaufer** [comment]
Summary from conversation with Julian (edited for clarity):

Break up the current Canopy Cover workflow into separate apps - ignoring ODM for now & assuming Orthomosaic images are the starting point. The following shows what the breaking up the workflow would look like (building off previous conversations) with each SciF app being a Makeflow workflow:
1. If the user doesn't have GeoJSON plots, they use SciF apps to convert from Shapefile or BETYdb to GeoJSON
2. They then run a workflow that masks their ortho image and clips both the ortho images (original & masked) to plots
3. Since there results of clipping can vary, they run a clipped image discovery app to locate all the clipped images, producing a JSON file to be used for the algorithm workflows
4. They run one or more algorithm apps, such as greenness indices, using makeflow and the JSON file from step 3. to generate plot level CSV information and have it merged into a top level CSV file

Regarding: For algorithms, how does our group and researchers create the app and Workflow for each algorithm?
1. Make a SCIF recipe for a tool
2. Add tool to a Makeflow .jx file

Some unknowns:
1. Is there an issue with using a Makeflow workflow outside of the SciF container to run the Makeflow workflows **within** the SciF container
2. How does this approach work with/on an HPC

Other items:
- Provide documentation: how to make a SCIF recipe for Algorithms
- Provide documentation: how to add Algorithm tool to a Makeflow .jx file to be incorporated as an app in SciF Docker image
- Provide documentation: how to build a new SciF Docker image and test (might be able to use existing documentation)

**Comment by Chris-Schnaufer** [comment]
PR: https://github.com/AgPipeline/drone-makeflow/pull/27 (Additional PR's related to this effort are expected)

This resides on branch: https://github.com/AgPipeline/drone-makeflow/tree/split_workflows
GitHub Action result: https://github.com/AgPipeline/drone-makeflow/actions/runs/325003252

Waiting on issue before making a n official release: https://github.com/AgPipeline/issues-and-projects/issues/337
